src/cockpit/channels/filesystem.py

```diff
@@ -118,15 +117,28 @@ class FsReplaceChannel(Channel):
     _path = None
     _tag = None
     _tempfile = None
+    _temppath = None
 
     def do_open(self, options):
         self._path = options.get('path')
         self._tag = options.get('tag')
 
     def do_data(self, data):
         if self._tempfile is None:
-            dirname, basename = os.path.split(self._path)
-            self._tempfile = tempfile.NamedTemporaryFile(dir=dirname, prefix=f'.{basename}-', delete=False)
+            for i in range(1000):
+                self._temppath = f'{self._path}.cockpit-tmp.{i}'
+                try:
+                    fd = os.open(self._temppath, os.O_CREAT | os.O_WRONLY | os.O_EXCL, 0o666)
+                    break
+                except FileExistsError:
+                    continue
+                except OSError as ex:
+                    raise ChannelError('internal-error', message=str(ex))
```

Comment:
At least one reasonable failure here is also permission errors (ie: not being allowed to create files in that directory).  Is that worth reporting separately?  Certainly doesn't seem like an "internal error".

Could also consider `ENOSPC` as a user-meaningful message.